### PR TITLE
Introduce Conformer

### PR DESCRIPTION
**Pull request by hwangjeff**
Adds implementation of Conformer module.

Adapted from @sravyapopuri388's implementation for fairseq at https://github.com/fairinternal/fairseq-py/pull/2770.

### CODE REVIEW

**Comment by kobenaxie** [comment]
support streaming conformer or not ?

**Comment by kobenaxie** [comment]
https://github.com/pytorch/audio/blob/0a701058b432dd602bba3461866bfb3c3a352e04/torchaudio/prototype/rnnt_decoder.py#L156 what does 4 mean here ?

**Comment by hwangjeff** [comment]
@kobenaxie this pr only covers non-streaming conformer; we may introduce a streaming variant later on. in the meantime, emformer (https://pytorch.org/audio/main/prototype.html#emformer) might be of interest to you as it's streaming-capable

re: the decoder — currently, the decoder assumes that the first four tokens in the token vocabulary are the special symbols `<bos>`, `<pad>`, `<eos>`, and `<unk>`. this is the basis for the line you referenced. we will generalize this behavior in a future pr

**Comment by kobenaxie** [comment]
@hwangjeff   Thank you for your reply, and i find a duplicate declaration here , https://github.com/pytorch/audio/blob/0a701058b432dd602bba3461866bfb3c3a352e04/torchaudio/prototype/emformer.py#L742

File: test/torchaudio_unittest/prototype/conformer_test_impl.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+                depthwise_conv_kernel_size=31,
+                dropout=0.1,
+            )
+            .to(device=self.device, dtype=self.dtype)
+            .eval()
+        )
+        return conformer
+
+    def _gen_inputs(self, input_dim, batch_size, num_frames):
+        lengths = torch.randint(1, num_frames, (batch_size,)).to(
```
[the remark this answers is not in the document]
Comment (hwangjeff) [reply test/torchaudio_unittest/prototype/conformer_test_impl.py:27]: i keep forgetting about this — thanks

**Review by mthrok — APPROVED** [review]

**Pull request closed without merging**